modules/lvgl/input/lvgl_common_input.c

```diff
@@ -26,8 +26,16 @@ static void lvgl_input_read_cb(lv_indev_drv_t *drv, lv_indev_data_t *data)
 {
 	const struct device *dev = drv->user_data;
 	const struct lvgl_common_input_config *cfg = dev->config;
+	struct lvgl_common_input_data *common_data = dev->data;
+
+	if (k_msgq_get(cfg->event_msgq, data, K_NO_WAIT) != 0) {
+		memcpy(data, &common_data->previous_event, sizeof(lv_indev_data_t));
+		data->enc_diff = 0; /* For encoders, clear last movement */
```

Comment:
Do you think we should only do this for encoder events? I know it does not matter but I'm thinking if the struct changes to a union this may introduce some nasty bug, may be worth playing safe and check `indev_drv->type here`

Reply:
Sure safeguarding never a bad idea :^)